From d7db32d45ebcd1ed49d94345a48fb4124e48064e Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 25 Dec 2022 16:51:23 +0200
Subject: [PATCH] [SW-116214] gaudi: security: fix hbw rrs masks

RRs masks are covering only 46 bits, while the actual address space is
50 bits.
This results in higher address space being protected while it should not
be.
Fix this by extending the HBW masks to 50 bits.

Change-Id: I2c4b42ebb7d397c87c5b8f85d95bf915b37d931b
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index 2838338..ff569dd 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -1747,7 +1747,7 @@
 	set_prot_bits_range_all(prot_base, PROT_SIZE_MAX);
 
 	/* Define mask of 32MB, so we close the SRAM in one shot */
-	preboot_secure_hbw_tpc3_seclvl_rr(SDRAM_BASE, 0x3FFFFE000000, true);
+	preboot_secure_hbw_tpc3_seclvl_rr(SDRAM_BASE, 0x3FFFFFE000000, true);
 }
 
 static void preboot_secure_hbw_tpc3_seclvl_open(void)
@@ -1785,25 +1785,25 @@
 	preboot_secure_hbw_sram();
 
 	/* SRAM Reduct. RR 0x7F_F140_0000 - 0x7F_F180_0000 (4MB) */
-	preboot_secure_hbw_rr(HBW_RR_SRAM_R_BGN, 0x7FF1400000, 0x3FFFFFC00000);
+	preboot_secure_hbw_rr(HBW_RR_SRAM_R_BGN, 0x7FF1400000, 0x3FFFFFFC00000);
 
 	/* SRAM Reduct. RR 0x7F_F180_0000 - 0x7F_F200_0000 (8MB) */
-	preboot_secure_hbw_rr(HBW_RR_SRAM_R_1, 0x7FF1800000, 0x3FFFFF800000);
+	preboot_secure_hbw_rr(HBW_RR_SRAM_R_1, 0x7FF1800000, 0x3FFFFFF800000);
 
 	/* SRAM Reduct. RR 0x7F_F200_0000 - 0x7F_F280_0000 (8MB) */
-	preboot_secure_hbw_rr(HBW_RR_SRAM_R_END, 0x7FF2000000, 0x3FFFFF800000);
+	preboot_secure_hbw_rr(HBW_RR_SRAM_R_END, 0x7FF2000000, 0x3FFFFFF800000);
 
 	/* STM RR 0x7F_F400_0000 - 0x7F_F800_0000 (64MB) */
-	preboot_secure_hbw_rr(HBW_RR_STM, 0x7FF4000000, 0x3FFFFC000000);
+	preboot_secure_hbw_rr(HBW_RR_STM, 0x7FF4000000, 0x3FFFFFC000000);
 
 	/* SPI Flash RR 0x7F_F800_0000 - 0x7F_FA00_0000 (32MB) */
-	preboot_secure_hbw_rr(HBW_RR_SPI_FLASH, HL_SPI_BASE, 0x3FFFFE000000);
+	preboot_secure_hbw_rr(HBW_RR_SPI_FLASH, HL_SPI_BASE, 0x3FFFFFE000000);
 
 	/* Scratchpad SRAM RR 0x7F_FBFE_0000 - 0x7F_FBFF_0000 (64KB) */
-	preboot_secure_hbw_rr(HBW_RR_SP_SRAM, 0x7FFBFE0000, 0x3FFFFFFF0000);
+	preboot_secure_hbw_rr(HBW_RR_SP_SRAM, 0x7FFBFE0000, 0x3FFFFFFFF0000);
 
 	/* PCIe FW SRAM RR 0x7F_FBFF_0000 - 0x7F_FBFF_8000 (32KB) */
-	preboot_secure_hbw_rr(HBW_RR_PCIE_FW, 0x7FFBFF0000, 0x3FFFFFFF8000);
+	preboot_secure_hbw_rr(HBW_RR_PCIE_FW, 0x7FFBFF0000, 0x3FFFFFFFF8000);
 
 	/* HBM RR 0 - 32GB */
 	/* Keep RR_0 last, as it replaces the BTL settings */
